akka-stream/src/main/scala/akka/stream/javadsl/SubFlow.scala

```diff
@@ -163,7 +163,36 @@ class SubFlow[-In, +Out, +Mat](delegate: scaladsl.SubFlow[Out, Mat, scaladsl.Flo
    * '''Cancels when''' downstream cancels
    */
   def mapConcat[T](f: function.Function[Out, java.lang.Iterable[T]]): SubFlow[In, T, Mat] =
-    new SubFlow(delegate.mapConcat { elem ⇒ Util.immutableSeq(f(elem)) })
+    new SubFlow(delegate.statefulMapConcat { () ⇒ elem ⇒ Util.immutableSeq(f(elem)) })
+
+  /**
+    * Transform each input element into an `Iterable` of output elements that is
+    * then flattened into the output stream. Transformation is stateful
+    * meaning it's taking into account previous messages (hysteresis).
+    *
+    * Make sure that the `Iterable` is immutable or at least not modified after
+    * being used as an output sequence. Otherwise the stream may fail with
+    * `ConcurrentModificationException` or other more subtle errors may occur.
+    *
+    * The returned `Iterable` MUST NOT contain `null` values,
+    * as they are illegal as stream elements - according to the Reactive Streams specification.
+    *
+    * '''Emits when''' the mapping function returns an element or there are still remaining elements
+    * from the previously calculated collection
+    *
+    * '''Backpressures when''' downstream backpressures or there are still remaining elements from the
+    * previously calculated collection
+    *
+    * '''Completes when''' upstream completes and all remaining elements has been emitted
+    *
+    * '''Cancels when''' downstream cancels
+    */
+  def statefulMapConcat[T](f: function.Creator[function.Function[Out, java.lang.Iterable[T]]]): SubFlow[In, T, Mat] =
```

Comment:
Is the intention to create `stateful` variants of other things also, such as `statefulMap` and `statefulFilter`, or is `statefulMapConcat` the universal `stateful` tool?